### PR TITLE
Fix vmTemplate latestVersion isn't updated correctly when set the new created templateVersion as default

### PR DESCRIPTION
**Pull request by aiwantaozi**
genclient:skipVerbs=update just control whether we generate update clientset or not, can't forbid the PUT API, so we get this back for testing.

Issue: 
https://github.com/rancher/harvester/issues/121
https://github.com/rancher/harvester/issues/118

### CODE REVIEW

**Review by guangbochen — APPROVED** [review]

File: pkg/api/vmtemplate/template_store.go [hunk 1]
```diff
@@ -4,24 +4,30 @@ import (
...
 	}
+
+	currentTp, err := convert.EncodeToMap(tp)
+	if err != nil {
+		return types.APIObject{}, apierror.NewAPIError(validation.ServerError, err.Error())
+	}
+
+	//not allow UI update the status
+	data.Data().SetNested(currentTp["statue"], "status")
+	data.Data().SetNested(datautil.GetValueN(currentTp, "metadata", "resourceVersion"), "metadata", "resourceVersion")
```
Comment (gitlawr) [inline pkg/api/vmtemplate/template_store.go]: Can we use [Status subresource](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#status-subresource) for this purpose.

Comment (gitlawr) [inline pkg/api/vmtemplate/template_store.go:31]: Can we use [Status subresource](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#status-subresource) for this purpose.

**Comment by aiwantaozi** [comment]
updated, this need to update the crd virtualmachinetemplates.harvester.cattle.io

File: pkg/controller/master/template/template_version_controller.go [hunk 2]
```diff
@@ -70,39 +69,32 @@ func (h *templateVersionHandler) OnChanged(key string, tv *apisv1alpha1.VirtualM
 
 	//set version
 	if !apisv1alpha1.VersionAssigned.IsTrue(copyObj) {
-		existLatestVersion, err := h.getTemplateLatestVersion(tv.Namespace, tv.Name, tv.Spec.TemplateID)
+		selector := fmt.Sprintf("metadata.name!=%s", tv.Name)
+		existLatestVersion, _, err := getTemplateLatestVersion(tv.Namespace, tv.Spec.TemplateID, selector, h.templateVersions)
```
Comment (gitlawr) [inline pkg/controller/master/template/template_version_controller.go]: It looks obscure to pass such a selector string as the function parameter. Why do we want to extract it out?

**Review by gitlawr — APPROVED** [review]
LGTM

**Pull request merged**